### PR TITLE
Sentry: Calm the fuck down!

### PR DESCRIPTION
**Pull request by kayone**

### CODE REVIEW

File: src/NzbDrone.Common/Instrumentation/Sentry/SentryTarget.cs [hunk 1]
```diff
@@ -52,6 +54,37 @@ public SentryTarget(string dsn)
             _client.Tags.Add("culture", Thread.CurrentThread.CurrentCulture.Name);
             _client.Tags.Add("branch", BuildInfo.Branch);
             _client.Tags.Add("version", BuildInfo.Version.ToString());
+
+            _debounce = new SentryDebounce();
+        }
+
+
+        private List<string> GetFingerPrint(LogEventInfo logEvent)
```
Comment (Taloth) [inline src/NzbDrone.Common/Instrumentation/Sentry/SentryTarget.cs]: Why not just return a string? I think you can put the string.Join in this method.

File: src/NzbDrone.Common/Instrumentation/Sentry/SentryTarget.cs [hunk 2]
```diff
@@ -52,6 +54,37 @@ public SentryTarget(string dsn)
...
+            var fingerPrint = new List<string>
+            {
+                logEvent.Level.Ordinal.ToString(),
+            };
+
+            var lineNumber = "";
+
+            if (logEvent.StackTrace != null)
+            {
+                var stackFrame = logEvent.StackTrace.GetFrame(logEvent.UserStackFrameNumber);
```
Comment (Taloth) [inline src/NzbDrone.Common/Instrumentation/Sentry/SentryTarget.cs]: As mentioned, get rid of the stackframe&line number stuff. Use LogEventInfo.LoggerName + LogEventInfo.Message instead.

File: src/NzbDrone.Common/Instrumentation/Sentry/SentryDebounce.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NzbDrone.Common.Instrumentation.Sentry
+{
+    public class SentryDebounce
+    {
+        private readonly Dictionary<string, DateTime> _dictionary;
```
Comment (Taloth) [inline src/NzbDrone.Common/Instrumentation/Sentry/SentryDebounce.cs]: Why not use Cached? It has a built in expiry iirc.

Also, maybe have a counter on the number of events? If that's relevant to Sentry. (Not a priority, do it later when it's stabilized)

File: src/NzbDrone.Common/Instrumentation/Sentry/SentryTarget.cs [hunk 4]
```diff
@@ -52,6 +54,37 @@ public SentryTarget(string dsn)
...
+                {
+                    lineNumber = $"#{stackFrame.GetFileLineNumber()}";
+                }
+            }
+
+            fingerPrint.Add(logEvent.LoggerName + lineNumber);
+
+            if (logEvent.Exception != null)
+            {
+                fingerPrint.Add(logEvent.Exception.GetType().Name);
```
Comment (Taloth) [inline src/NzbDrone.Common/Instrumentation/Sentry/SentryTarget.cs]: Maybe include the InnerException type as well if present? Helps segregate between different causes.

File: src/NzbDrone.Common/Instrumentation/Sentry/SentryDebounce.cs (hunk 3 above)
Comment (kayone) [inline src/NzbDrone.Common/Instrumentation/Sentry/SentryDebounce.cs]: sentry doesn't it's own internal count.

**Pull request merged**